frontend/catalyst/qfunc.py

```diff
@@ -257,7 +257,9 @@ def processing_fn(results):
     total_shots = get_device_shots(dev)
 
     new_dev = copy(dev)
-    if isinstance(new_dev, qml.devices.LegacyDevice):
+    if isinstance(new_dev, qml.devices.LegacyDeviceFacade):
+        new_dev.target_device.shots = 1
+    elif isinstance(new_dev, qml.devices.LegacyDevice):
```

Comment:
This check is redundant. Since the facade will ALWAYS exist around legacy devices, we don't want any special handling for Legacy devices in the wild, as this should be considered a bug.
This branch should be deleted.

Reply:
Thanks @Shiro-Raven for the comment! I was trying to be conservative so it works in a wild case where the target_device is extracted from the facade and used separately, but seems like we don't want that to happen. removed the branch

Reply:
We tried our best to cover all potential insertions of a device into a circuit execution such that it always gets caught and wrapped with the facade. If a scenario shows up in the future where a user was able to somehow circumvent all the checks we added, I think it would be better to handle it in PL instead of it being silently handled in other libraries.

Reply:
Actually removing that check causes test failures for kokkos. Since kokkos CI job is using version 0.37.0 of lightning, I think maybe it is best to bring the check back and have TODO to remove it after the next release. What do you think @dime10?

Reply:
Shouldn't we be using the dev version of Kokkos? Where does this happen?

Reply:
you are right! sorry I looked at the wrong place. It actually uses the dev version. so we are good.